### ISSUE TITLE
Issue in @clerk/clerk-expo@2.2.12

### ISSUE THREAD

**Issue by hyochan** [clerk/javascript#4178]
### Preliminary Checks

- [X] I have reviewed the documentation: https://clerk.com/docs

- [X] I have searched for existing issues: https://github.com/clerk/javascript/issues #4177

- [X] I have not already reached out to Clerk support via email or Discord (if you have, no need to open an issue here)

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Clerk. Please ask questions in our Discord community: https://clerk.com/discord.

### Reproduction

https://github.com/clerk/clerk-expo-quickstart

### Publishable key

pk_test_c2VjdXJlLWdvcmlsbGEtOTUuY2xlcmsuYWNjb3VudHMuZGV2JA

### Description

**Steps to reproduce:**

1. Clone https://github.com/clerk/clerk-expo-quickstart
2. Install `"@clerk/clerk-expo": "^2.2.12",`
3. Run expo app

**Expected behavior:**
Run without error

**Actual behavior:**
Error occurs like below. Works fine when downgrading to `2.0.0`.

![Simulator Screenshot - iPhone 15 - 2024-09-17 at 15 42 20](https://github.com/user-attachments/assets/08cc79ef-6b64-44ae-9a02-ca88c040d000)
```
Unable to resolve "@clerk/shared/authorization" from "node_modules/@clerk/clerk-react/dist/internal.js"
```

It also works in `2.2.10` and not working from `2.2.11`.
Below changes might have some issue https://github.com/clerk/javascript/compare/%40clerk/clerk-expo%402.2.10...%40clerk/clerk-expo%402.2.11

### Environment

```Shell
System:
    OS: macOS 15.0
    CPU: (8) arm64 Apple M1
    Memory: 293.44 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v20.10.0/bin/yarn
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
    bun: 1.1.21 - ~/.nvm/versions/node/v20.10.0/bin/bun
  Browsers:
    Chrome: 128.0.6613.138
    Safari: 18.0
  npmPackages:
    @babel/core: ^7.20.0 => 7.24.5 
    @clerk/clerk-expo: ^2.2.12 => 2.2.12 
    @expo/vector-icons: ^14.0.2 => 14.0.2 
    @react-navigation/native: ^6.0.2 => 6.1.17 
    @types/jest: ^29.5.12 => 29.5.12 
    @types/react: ~18.2.45 => 18.2.79 
    @types/react-test-renderer: ^18.0.7 => 18.3.0 
    expo: ~51.0.31 => 51.0.32 
    expo-constants: ~16.0.2 => 16.0.2 
    expo-font: ~12.0.10 => 12.0.10 
    expo-linking: ~6.3.1 => 6.3.1 
    expo-router: ~3.5.23 => 3.5.23 
    expo-secure-store: ~13.0.2 => 13.0.2 
    expo-splash-screen: ~0.27.5 => 0.27.5 
    expo-status-bar: ~1.12.1 => 1.12.1 
    expo-system-ui: ~3.0.7 => 3.0.7 
    expo-web-browser: ~13.0.3 => 13.0.3 
    jest: ^29.2.1 => 29.7.0 
    jest-expo: ~51.0.4 => 51.0.4 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    react-native: 0.74.5 => 0.74.5 
    react-native-gesture-handler: ~2.16.1 => 2.16.2 
    react-native-reanimated: ~3.10.1 => 3.10.1 
    react-native-safe-area-context: 4.10.5 => 4.10.5 
    react-native-screens: 3.31.1 => 3.31.1 
    react-native-web: ~0.19.10 => 0.19.11 
    react-test-renderer: 18.2.0 => 18.2.0 
    typescript: ~5.3.3 => 5.3.3
```

**Comment by yessineZ** [comment]
just downgrade the version of the clerk to @2.2.10

**Comment by hyochan** [comment]
@yessineZ Yes that is how I solved it! However, very curious why it can't resolve the package when `@clerk/shared` package in in `@clerk/react` pacakge.
I suspect regression might happen in.https://github.com/clerk/javascript/pull/4118

**Comment by LekoArts** [comment]
This should be fixed with https://github.com/clerk/javascript/pull/4172 - once https://github.com/clerk/javascript/pull/4179 is merged a new version of `@clerk/expo` will be released.

**Comment by LekoArts** [comment]
This should be fixed with `@clerk/expo` version `2.2.13`

**Comment by clerk-cookie** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
